os/tools/mkheader.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+############################################################################
+#
+# Copyright 2019 Samsung Electronics All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+############################################################################
+import os
+import sys
+import hashlib
+############################################################################
+# header information
+# / header size (2byte) / binary type (2byte) / binary size (4byte)
+# / Binary name (16byte) / Binary version (12byte)
+# / Kernel version (8byte) / Jump address (4byte)
+# header size is 48byte.
+#
+# parameter information
+# argv[1] is filepath
+# argv[2] is file type
+# argv[3] is Kernel version
+# argv[4] is Binary name
+# argv[5] is Binary version
+############################################################################
+
+header_size = 48
+file_path  = sys.argv[1]
+binary_type = sys.argv[2]
+kernel_ver = sys.argv[3]
+binary_name = sys.argv[4]
+binary_ver = sys.argv[5]
+
+binary_ver_point = 24
```

Comment:
I think it is better to define size values of each variable, not calculated offset.

Reply:
some codes would be changed, for example, in line 69
fp.write(binary_name);
offset += binary_name_size;

fp.seek(offset, 0);
fp.write(binary_ver);

Reply:
I modified it.